src/app/app.tsx

```diff
@@ -1,41 +1,73 @@
-import React, { ReactNode, useEffect } from 'react';
+import React, { useEffect } from 'react';
 import { AppHeader } from '../app-header/app-header';
-import { AppErrorDisplay } from '../errors/app-error-display';
 import { useMonitoring } from '../monitoring/monitoring-provider';
-import { ReportingConfigLoadingIndicator } from '../reporting-config/reporting-config-loading-indicator';
-import { useReportingConfigLoad } from '../reporting-config/use-reporting-config';
-import { ReportingFlow } from '../reporting-flow/reporting-flow';
-import { useInitialStateFromUrl } from '../url-history/hooks';
 import styles from './app.module.css';
 import { AppErrorBoundary } from '../errors/app-error-boundary';
+import { selectActivePage } from '../active-page/active-page-slice';
+import { useAppSelector } from './hooks';
+import { DuplicateSearchingPage } from '../duplicate-searching-page/duplicate-searching-page';
+import { ReportingFlowPage } from '../reporting-flow-page/reporting-flow-page';
+import { useAppDataHydration } from './use-app-data-hydration';
+import { AnimatedEllipsis } from '../common/components';
 
 export function App() {
-	const { loadStatus, error } = useReportingConfigLoad();
 	const monitoringClient = useMonitoring();
-	useInitialStateFromUrl();
-
 	useEffect( () => {
 		monitoringClient.analytics.recordEvent( 'page_view' );
 	}, [ monitoringClient.analytics ] );
 
-	let mainDisplay: ReactNode;
-	if ( loadStatus === 'empty' || loadStatus === 'loading' ) {
-		mainDisplay = <ReportingConfigLoadingIndicator />;
-	} else if ( loadStatus === 'loaded' ) {
-		mainDisplay = <ReportingFlow />;
-	} else {
-		monitoringClient.logger.error( 'Error occurred when loading the reporting config', {
-			error: error,
-		} );
-		mainDisplay = <AppErrorDisplay />;
-	}
-
 	return (
 		<div className={ styles.appMain }>
 			<AppHeader />
 			<AppErrorBoundary>
-				<main className={ styles.appMain }>{ mainDisplay }</main>
+				<main className={ styles.appMain }>
+					<MainDisplay />
+				</main>
 			</AppErrorBoundary>
 		</div>
 	);
 }
+
+function MainDisplay() {
+	// Doing all the main data hydration here so any truly unexpected errors (should be extremely rare!) bubble up to the error boundary.
+	// In general though, all we care about here is whether they hydration is done or not.
+	// We defer any expected error handling (e.g. request failure, bad configuration) to the child components.
+	// This allows as much of the app to be usable as possible, even if some parts are not working.
+	const ready = useAppDataHydration();
+	const activePage = useAppSelector( selectActivePage );
+
+	if ( ! ready ) {
+		return <AppLoadingIndicator />;
+	}
+
+	if ( activePage === 'duplicateSearching' ) {
```

Comment:
That's a good point, and we definitely can! But I think we should be protected by the fact that we're using TypeScript and the type isn't `string`, but rather a union type of two specific strings `'duplicateSearching' | 'reportingFlow'`. So if you try to set it to a different string, it actually will cause a type error and the app won't build!

We use this pattern in a handful of different places throughout the app (for example, filter values, the issue type, etc.). I think in general, I'm more of a fan of using union types than constants because they end up requiring less code and actually block the site from being built if something is wrong.

Let me know what you think! 😄